pkg/controller/job/job_controller_test.go

```diff
@@ -3045,13 +3095,138 @@ func TestSyncJobWithJobPodFailurePolicy(t *testing.T) {
 					},
 				},
 			},
-			wantStatusActive: 1, // This is a replacement Pod: the terminating Pod is neither active nor failed.
+			wantStatusActive:      1, // This is a replacement Pod: the terminating Pod is neither active nor failed.
+			wantStatusTerminating: pointer.Int32(1),
+		},
+		"default value for podreplacementpolicy is Failed when not specifying PodReplacementPolicy": {
```

Comment:
Quite a big test case as for just testing defaulting, especially as we don't directly test that the value was defaulted. Maybe a more behavior-oriented comment "terminating pods have delayed recreation until terminal when PodFailurePolicy is used"?

I assume we test the pods aren't recreated by `wantStatusActive=0`, but please double check.

Reply:
We do correctly check `wantStatusActive=0` now.